api/turing/api/pod_log_api.go

```diff
@@ -176,91 +163,26 @@ func (c PodLogController) ListEnsemblingJobPodLogs(_ *http.Request, vars Request
 	return Ok(logs)
 }
 
-func getComponentType(defaultComponentType string, vars RequestVars, allowedTypes map[string]struct{}) (string, error) {
-	componentType, _ := vars.get("component_type")
-	if componentType == "" {
-		return defaultComponentType, nil
-	}
-
-	if _, ok := allowedTypes[componentType]; ok {
-		return componentType, nil
-	}
-
-	return "", fmt.Errorf("Invalid component type '%s'", componentType)
-}
-
-type ListEnsemblingPodLogsOptions struct {
+type podLogOptions struct {
 	ProjectID *models.ID `schema:"project_id" validate:"required"`
-	ID        *models.ID `schema:"job_id" validate:"required"`
+	Previous  bool       `schema:"previous"`
+	HeadLines *int64     `schema:"head_lines" validate:"omitempty,gte=0"`
+	TailLines *int64     `schema:"tail_lines" validate:"omitempty,gte=0"`
+	SinceTime *time.Time `schema:"since_time"`
```

Comment:
What is the default time format? Is it `RFC3339`? Can it also recognize millis (example: `2006-01-02T15:04:05.244Z`)?

Reply:
Yes it can, the reason why I did not put the validator is because it will fail the parsing stage, which is done before the validation.